src/Components/Nav/Nav.jsx

```diff
@@ -1,26 +1,33 @@
 import React from "react";
 import { Link } from "react-router-dom";
+import { useWishlist } from "../../Context/wishlist";
 import "./Nav.css";
 
 function Nav() {
+  const { state } = useWishlist();
   return (
-      <header>
-        <div className="header-brand">
-          <Link to="/" className="btn-link">
-            Always Shopping
-          </Link>
-        </div>
-        <div className="nav-right center-div">
-          <Link to="/wishlist" className="badge-icon-wrapper center-div p-1 btn-link">
-            <i className="far fa-heart"></i>
-            <span className="badge-icon center-div">1</span>
-          </Link>
-          <Link to="/cart" className="badge-icon-wrapper center-div p-1 btn-link">
-            <i className="fas fa-cart-plus"></i>
-            <span className="badge-icon">6</span>
-          </Link>
-        </div>
-      </header>
+    <header>
+      <div className="header-brand">
+        <Link to="/" className="btn-link">
+          Always Shopping
+        </Link>
+      </div>
+      <div className="nav-right center-div">
+        <Link
+          to="/wishlist"
+          className="badge-icon-wrapper center-div p-1 btn-link"
+        >
+          <i className="far fa-heart"></i>
+          <span className="badge-icon center-div">
+            {state.wishlist?.length}
```

Comment:
You should be using optional chaining only if you are unsure about the data you are getting. Here since your state is defined by your own code, it does not make sense to check for availability of wishlist key inside state object

Reply:
> Code looks clean and modular I discovered the concept of `optional chaining` from your code

Thank you maitra

Reply:
Yes, make sense. Thanks, I will refactor it.